airbyte-cdk/java/airbyte-cdk/typing-deduping/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/LoggingUtil.kt

```diff
@@ -0,0 +1,11 @@
+package io.airbyte.integrations.base.destination.typing_deduping
+
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+/**
+ * Extends all classes to lazily initialize a Slf4j [Logger]
+ */
+fun <R: Any> R.logger(): Lazy<Logger> {
+    return lazy { LoggerFactory.getLogger(this.javaClass) }
+}
```

Comment:
On the platform side we use [Kotlin-Logging](https://github.com/oshai/kotlin-logging) for logging configuration.